crates/iceberg/src/spec/sort.rs

```diff
@@ -133,6 +133,14 @@ impl SortOrder {
     pub fn is_unsorted(&self) -> bool {
         self.fields.is_empty()
     }
+
+    /// Set the order id for the sort order
+    pub fn with_order_id(&self, order_id: i64) -> SortOrder {
```

Comment:
An API named `with_xxx` that takes a ref might be a bit confusing. Is there any context for this?

Reply:
You are right, owned version is much cleaner. Fixed in latest commit :)